helix-term/src/commands.rs

```diff
@@ -5157,122 +5176,239 @@ pub fn completion(cx: &mut Context) {
 }
 
 // comments
-type CommentTransactionFn = fn(
-    line_token: Option<&str>,
-    block_tokens: Option<&[BlockCommentToken]>,
-    doc: &Rope,
+
+/// Perform a `Transaction` to toggle comments
+type CommentTransaction = fn(
+    text: &Rope,
     selection: &Selection,
+    doc_line_token: Option<&str>,
+    doc_block_tokens: Option<&[BlockCommentToken]>,
+    syntax: Option<&Syntax>,
+    loader: &syntax::Loader,
```

Comment:
With #12275 I think this became ambiguous. Here we can use `helix_core::syntax::Loader` instead and remove the `use` of `helix_core::syntax::self`